Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm

```diff
@@ -103,42 +103,42 @@ - (void)invalidate
 - (void)streamDataParser:(AVStreamDataParser *)streamDataParser didParseStreamDataAsAsset:(AVAsset *)asset
 {
     ASSERT_UNUSED(streamDataParser, streamDataParser == _parser);
-    _parent->didParseStreamDataAsAsset(asset);
+    Ref { *_parent }->didParseStreamDataAsAsset(asset);
```

Comment:
Hm... I guess we're assuming all these will be non-null because otherwise we would have a UAF?
Should we consider using WeakRef in that case?

Reply:
The reason I cannot use a WeakRef is because of how ObjC objects are initialized, from an `init` function. WeakRef complains that it doesn't have a default constructor.

Reply:
Oh I see, that's unfortunate :(

Reply:
```
/Volumes/Work/WebKit/OpenSource/Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:76:17: error: no matching constructor for initialization of 'WeakRef<WebCore::SourceBufferParserAVFObjC>'
   76 | @implementation WebAVStreamDataParserListener
```